extension/chrome/settings/setup/setup-create-key.ts

```diff
@@ -35,7 +40,56 @@ export class SetupCreateKeyModule {
       /* eslint-enable @typescript-eslint/naming-convention */
       const keyAlgo = this.view.clientConfiguration.getEnforcedKeygenAlgo() || ($('#step_2a_manual_create .key_type').val() as KeyAlgo);
       const keyIdentity = await this.createSaveKeyPair(opts, keyAlgo);
-      if (this.view.clientConfiguration.canBackupKeys()) {
+      if (this.view.clientConfiguration.getPublicKeyForPrivateKeyBackupToDesignatedMailbox()) {
+        const adminPubkey = this.view.clientConfiguration.getPublicKeyForPrivateKeyBackupToDesignatedMailbox();
+        if (adminPubkey) {
+          const msgEncryptionKey = await KeyUtil.parse(adminPubkey);
+          const destinationEmail = msgEncryptionKey.emails[0];
+          try {
+            const privateKey = await KeyStore.get(this.view.acctEmail);
+            const primaryKeyId = await privateKey[0].id;
+            const primaryUserId = privateKey[0].emails![0];
+            await this.view.backupUi.initialize({
+              acctEmail: this.view.acctEmail,
+              action: 'setup_automatic',
+              keyIdentity,
+              onBackedUpFinished: async () => {
+                $('pre.status_details').remove();
+                $('#backup-template-container').remove();
+                await this.view.finalizeSetup();
+                await this.view.setupRender.renderSetupDone();
+              },
+            });
+            const parsedPrivateKey = await KeyUtil.parse(privateKey[0].private);
+            await OpenPGPKey.decryptKey(parsedPrivateKey, opts.passphrase);
+            const armoredPrivateKey = KeyUtil.armor(parsedPrivateKey);
```

Comment:
It looks like you armor encrypted key, as result of `OpenPGPKey.decryptKey` isn't used anywhere

Reply:
Hmm, maybe you're right, implementing test for restoration process (https://github.com/FlowCrypt/flowcrypt-browser/pull/5313#discussion_r1288361666) will be useful for checking it

Reply:
I understand. will do it.